gpio-falcon: fix recursive locking in .irq_set_type
this is called while already holding the "irq_desc_lock"

--- a/drivers/gpio/gpio-falcon.c
+++ b/drivers/gpio/gpio-falcon.c
@@ -165,12 +165,12 @@ static int falcon_gpio_irq_type(struct i
 	if ((type & (IRQ_TYPE_LEVEL_HIGH | IRQ_TYPE_LEVEL_LOW)) != 0) {
 		/* level triggered */
 		port_w32_mask(itop(d), 0, mask, GPIO_IRNCFG);
-		irq_set_chip_and_handler_name(d->irq,
+		irq_set_chip_handler_name_locked(d,
 			&falcon_gpio_irq_chip, handle_level_irq, "level");
 	} else {
 		/* edge triggered */
 		port_w32_mask(itop(d), mask, 0, GPIO_IRNCFG);
-		irq_set_chip_and_handler_name(d->irq,
+		irq_set_chip_handler_name_locked(d,
 			&falcon_gpio_irq_chip, handle_simple_irq, "edge");
 	}
 
